internal/keybase/sendMsg.go

```diff
@@ -2,32 +2,87 @@ package keybase
 
 import (
 	"bytes"
+	"encoding/json"
 	"fmt"
+	"io"
 	"net/http"
 	"os"
 
 	"github.com/chia-network/go-modules/pkg/slogs"
 )
 
+// Annotation represents the annotations in the alert
+type Annotation struct {
+	Title       string `json:"title"`
+	Description string `json:"description"`
+}
+
+// Alert represents a single alert
+type Alert struct {
+	Status      string     `json:"status"`
+	Annotations Annotation `json:"annotations"`
+}
+
+// WebhookMessage represents the message to be sent to the Keybase webhook
+type WebhookMessage struct {
+	Alerts []Alert `json:"alerts"`
+}
+
+// NewMessage creates and returns an instance of the WebhookMessage struct
+func NewMessage(status, title, description string) WebhookMessage {
+	alert := Alert{
+		Status: status,
+		Annotations: Annotation{
+			Title:       title,
+			Description: description,
+		},
+	}
+	return WebhookMessage{
+		Alerts: []Alert{alert},
+	}
+}
+
 // SendKeybaseMsg sends a message to a specified Keybase channel
-func SendKeybaseMsg(message string) error {
+func (msg *WebhookMessage) SendKeybaseMsg() error {
 	webhookURL := os.Getenv("KEYBASE_WEBHOOK_URL")
 	if webhookURL == "" {
 		return fmt.Errorf("KEYBASE_WEBHOOK_URL environment variable is not set")
 	}
 
-	resp, err := http.Post(webhookURL, "text/plain", bytes.NewBufferString(message))
+	authToken := os.Getenv("WEBHOOK_AUTH_SECRET_TOKEN")
+	if authToken == "" {
+		return fmt.Errorf("WEBHOOK_AUTH_SECRET_TOKEN environment variable is not set")
+	}
+
+	payload, err := json.Marshal(msg)
+	if err != nil {
+		slogs.Logr.Error("Error converting message to JSON", "error", err)
+		return err
+	}
+
+	client := &http.Client{}
```

Comment:
This is something I know Chris does pretty frequently. But you can actually create an init function that creates a global HTTP client that gets re-used. Re-using HTTP connections can reduce overhead and (albeit probably only slightly) improve performance. But it's something so simple to do that I usually recommend it, and I usually let it go when Chris makes a new HTTP client for each request when it's a script he runs for a short period of time in an Actions workflow. Since this will be running for a longer period of time in k8s, it makes sense imo to do the small things to make it more performant.